src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -544,18 +551,14 @@ CreateShardsOnWorkers(Oid distributedRelationId, List *shardPlacements,
 		 */
 		if (ShouldRecordRelationAccess() && useExclusiveConnection && partitionTable)
 		{
-			RelationShard *parentRelationShard = CitusMakeNode(RelationShard);
-			RelationShard *partitionRelationShard = CitusMakeNode(RelationShard);
-			List *relationShardList = NIL;
 			List *placementAccessList = NIL;
 
-			parentRelationShard->relationId = PartitionParentOid(distributedRelationId);
-			parentRelationShard->shardId =
-				ColocatedShardIdInRelation(parentRelationShard->relationId, shardIndex);
-			partitionRelationShard->relationId = distributedRelationId;
-			partitionRelationShard->shardId = shardId;
+			/* we only need to calculate this once per shardInterval, not placement */
+			if (relationShardList == NIL)
```

Comment:
This is always `true`. I don't know if there is an easy way to do what your comment says here.

Reply:
I think there is not way of doing that. Removing it.